### PR TITLE
etcd: use etcd-starter and bump to latest releases

### PR DESCRIPTION
**Pull request by crawford**
etcd-starter needs some work, but the rest of this should be good.

### CODE REVIEW

**Comment by crawford** [comment]
Depends on https://github.com/coreos/manifest/pull/60.

File: dev-db/etcd-starter/etcd-starter-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Header:$
+#
+
+EAPI=5
+CROS_WORKON_PROJECT="coreos/etcd-starter"
+CROS_WORKON_LOCALNAME="etcd-starter"
+CROS_WORKON_REPO="git://github.com"
+COREOS_GO_PACKAGE="github.com/coreos/etcd-starter"
+inherit coreos-doc coreos-go toolchain-funcs cros-workon
```
Comment (marineam) [inline dev-db/etcd-starter/etcd-starter-9999.ebuild]: Unless I'm missing something coreos-doc and toolchain-funcs are unused.

Comment (crawford) [inline dev-db/etcd-starter/etcd-starter-9999.ebuild]: Don't need coreos-doc

Comment (crawford) [inline dev-db/etcd-starter/etcd-starter-9999.ebuild]: Good timing.

Comment (crawford) [inline dev-db/etcd-starter/etcd-starter-9999.ebuild]: Fixed

**Comment by crawford** [comment]
Using etcd-starter 0.0.2.

**Comment by kelseyhightower** [comment]
@crawford lgtm

File: dev-db/etcd/etcd-0.4.8.ebuild [hunk 2]
```diff
@@ -37,14 +37,14 @@ src_compile() {
 }
```
Comment (marineam) [inline dev-db/etcd/etcd-0.4.8.ebuild]: `!etcd_protocols_2? ( !dev-db/etcd:2 )` needs to be updated to conflict with etcd-starter instead.

Comment (crawford) [inline dev-db/etcd/etcd-0.4.8.ebuild]: good catch

File: dev-db/etcd-starter/etcd-starter-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+DESCRIPTION="etcd-starter"
+HOMEPAGE="https://github.com/coreos/etcd-starter"
+SRC_URI=""
+
+LICENSE="Apache-2.0"
+SLOT="0"
+
+DEPEND=">=dev-lang/go-1.2"
+
```
Comment (marineam) [inline dev-db/etcd-starter/etcd-starter-9999.ebuild]: needs something like this:
```sh
RDEPEND="!dev-db/etcd:2[etcd_protocols_1]
    !dev-db/etcd:1[etcd_protocols_2]"
```

Basically this ebuild installs /usr/bin/etcd so it needs to conflict with the situations in which the other two ebuilds may install /usr/bin/etcd. Alternatively (if etcd-starter is ready) always install etcd-starter as /usr/bin/etcd so we can drop all this weird use flag logic to determine who installs /usr/bin/etcd

File: dev-db/etcd/etcd-0.4.8.ebuild (hunk 2 above)
Comment (marineam) [inline dev-db/etcd/etcd-0.4.8.ebuild]: On second thought putting the conflicts in etcd-starter would be simpler.

**Comment by crawford** [comment]
@marineam updated

File: dev-db/etcd/etcd-9999.ebuild [hunk 4]
```diff
@@ -24,7 +24,6 @@ SRC_URI=""
 
 LICENSE="Apache-2.0"
 SLOT="2"
-IUSE=""
```
Comment (marineam) [inline dev-db/etcd/etcd-9999.ebuild]: This line can be left alone.

Comment (crawford) [inline dev-db/etcd/etcd-9999.ebuild]: good to go

**Comment by marineam** [comment]
LGTM

**Pull request merged**